embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/network/logging/NetworkCaptureDataSourceImpl.kt

```diff
@@ -0,0 +1,42 @@
+package io.embrace.android.embracesdk.network.logging
+
+import io.embrace.android.embracesdk.Severity
+import io.embrace.android.embracesdk.arch.datasource.LogDataSourceImpl
+import io.embrace.android.embracesdk.arch.destination.LogEventData
+import io.embrace.android.embracesdk.arch.destination.LogEventMapper
+import io.embrace.android.embracesdk.arch.destination.LogWriter
+import io.embrace.android.embracesdk.arch.limits.NoopLimitStrategy
+import io.embrace.android.embracesdk.arch.schema.SchemaType
+import io.embrace.android.embracesdk.logging.InternalEmbraceLogger
+import io.embrace.android.embracesdk.payload.NetworkCapturedCall
+
+internal class NetworkCaptureDataSourceImpl(
+    private val logWriter: LogWriter,
+    logger: InternalEmbraceLogger
+) : NetworkCaptureDataSource,
+    LogEventMapper<NetworkCapturedCall>, LogDataSourceImpl(
+        destination = logWriter,
+        logger = logger,
+        limitStrategy = NoopLimitStrategy,
+    ) {
+
+    /**
+     * Creates a log with data from a captured network request.
+     *
+     * @param networkCapturedCall the captured network information
+     */
+    override fun logNetworkCapturedCall(networkCapturedCall: NetworkCapturedCall?) {
```

Comment:
Should we even allow a null here? We might just not want call this if the network capture object is null

Reply:
It was copied from the `LogService` interface but you are right that it doesn't make sense in the data source, changed to non-null